src/main/scala/Models.scala

```diff
@@ -1,21 +1,38 @@
+import spray.json._
+import DefaultJsonProtocol._
+import sangria.execution.deferred.HasId
+import sangria.schema._
+import sangria.macros.derive._
+
 case class Category(
   id: Int,
-  name: String
-) {
-  def styles(allStyles: List[Style]): List[Style] =
-    allStyles.filter(_.category_id == id)
+  name: String)
+
+object Category {
+  implicit val categoryHasId = HasId[Category, Int](_.id)
```

Comment:
`HasId` is a type-class needed for Fetch API to be able to cache and handle entities. It just provides an evidence/knowledge how to get an ID from your domain object (`Category` in this case)